From 5c8737c5c3170358024a4a969e1386cea15932f3 Mon Sep 17 00:00:00 2001
From: Michael Orlitzky <michael@orlitzky.com>
Date: Sun, 26 Apr 2020 09:56:34 -0400
Subject: [PATCH 1/1] src/tunetoom.c: delete duplicate definition of rp.

The "make tune-toom" command has started failing with gcc-10.x because
of its new default -fno-common behavior,

  * https://gcc.gnu.org/gcc-10/porting_to.html
  * https://wiki.gentoo.org/wiki/Gcc_10_porting_notes/fno_common

This leads to an error involving the FILE pointer "rp" that is declared
in global scope in both src/tunetoom.c and src/tuning-common.c. In this
case, the declaration in src/tunetoom.c is simply redundant: that file
includes src/tuning-common.h which already declares "rp" as extern.

Deleting the redeclaration in src/tunetoom.c makes the build succeed.
---
 src/tunetoom.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/tunetoom.c b/src/tunetoom.c
index 7553e0c..1140606 100644
--- a/src/tunetoom.c
+++ b/src/tunetoom.c
@@ -111,8 +111,6 @@ const char * gf2x_utoom_select_string[] = {
     [GF2X_SELECT_UNB_TC3U]  = "TC3U",
 };
 
-FILE *rp;
-
 void tunetoom(long tablesz)
 {
     long high, n;
-- 
2.24.1

